discretize/utils/meshutils.py

```diff
@@ -255,3 +257,438 @@ def ExtractCoreMesh(xyzlim, mesh, meshType='tensor'):
         raise Exception("Not implemented!")
 
     return actind, meshCore
+
+
+def mesh_builder_xyz(
+    xyz, h,
+    padding_distance=[[0, 0], [0, 0], [0, 0]],
+    base_mesh=None,
+    depth_core=None,
+    expansion_factor=1.3,
+    mesh_type='TENSOR'
+):
+    """
+        Function to quickly generate a Tensor of Tree mesh
+        given a cloud of xyz points, finest core cell size
+        and padding distance.
+        If a base_mesh is provided, the core cells will be centered
+        on the underlaying mesh to reduce interpolation errors.
+        The core extent is set by the bounding box of the xyz location.
+
+        :param numpy.ndarray xyz: Location points [n x dim]
+        :param list h: Cell size for the core mesh [1 x ndim]
+        :param list padding_distance: Padding distances
+            [[W,E], [N,S], [Down,Up]]
+        :param BaseMesh base_mesh: SimPEG mesh used to center the core mesh
+        :param float depth_core: Depth of core mesh below xyz
+        :param float expansion_factor: Expension factor for padding cells [1.3]
+        :param str mesh_type: Specify output mesh type: ["TENSOR"] or "TREE"
+        :return: Mesh of "mesh_type"
+
+    .. plot::
+
+        import discretize
+        import matplotlib.pyplot as plt
+        import numpy as np
+
+        xyLoc = np.random.randn(8,2)
+
+        mesh = discretize.utils.meshutils.mesh_builder_xyz(xyLoc, [0.1, 0.1],
+                                          depth_core=0.5,
+                                          padding_distance=[[1,2], [1,0]],
+                                          mesh_type='TENSOR',
+        )
+
+        # mesh.refine(5, finalize=True)
+        axs = plt.subplot()
+        mesh.plotImage(mesh.vol, grid=True, ax=axs)
+        axs.scatter(xyLoc[:,0], xyLoc[:,1], 15, c='w', zorder=3)
+        axs.set_aspect('equal')
+        plt.show()
+
+    """
+    if mesh_type not in ['TENSOR', 'TREE']:
+        raise Exception(
```

Comment:
I think raising a `ValueError` is the more appropriate exception here, instead of just throwing a general exception.